src/main/java/org/jenkinsci/plugins/workflow/cps/SnippetizerLink.java

```diff
@@ -75,32 +75,12 @@ public final String getDisplayUrl() {
 
         StaplerRequest req = Stapler.getCurrentRequest();
         if (req == null) {
-            return u;
+            throw new IllegalStateException("Can't get display URL without Stapler context.");
         }
 
         Item i = req.findAncestorObject(Item.class);
 
-        StringBuilder toAppend = new StringBuilder();
-
-        toAppend.append(req.getContextPath());
-
-        if (!req.getContextPath().endsWith("/")) {
-            toAppend.append("/");
-        }
-
-        if (i == null) {
-            toAppend.append(u);
-        } else {
-            toAppend.append(i.getUrl());
-
-            if (!i.getUrl().endsWith("/")) {
-                toAppend.append("/");
-            }
-
-            toAppend.append(u);
-        }
-
-        return toAppend.toString();
+        return req.getContextPath() + '/' + (i == null ? "" : i.getUrl()) + u;
```

Comment:
Wait, why did we remove the check for if the context path has a '/' on it?  We had an actual stage view bug that resulted from a similar issue.